### ISSUE TITLE
Keep viewport aspect ratio when resizing

### ISSUE THREAD

**Issue by blacktm** [simple2d/simple2d#28]
For some applications, like games, it's desirable to maintain the aspect ratio of the viewport, rather than stretch it to the full window width and height. This should be the default for new windows.

Furthermore, the viewport should grow proportionally as the window size increases. If there is extra width or height, the viewport should be centered, like this:

![a](https://cloud.githubusercontent.com/assets/887429/13275498/e40a99a0-da83-11e5-9f5b-7d30c693b135.jpg)

![b](https://cloud.githubusercontent.com/assets/887429/13275509/fcb2d472-da83-11e5-92fd-9b0d1c7d142d.jpg)

Also, add the ability to set viewport resizing to stretch mode.

**Comment by blacktm** [comment]
Here's a full program which simulates the desired behavior. The viewport isn't centered vertically or horizontally, need to figure that out.

```c
#include <simple2d.h>
#include <math.h>

S2D_Window *window;
int vp_w = 50;  // viewport width
int vp_h = 50;  // viewport height
int win_w = 0;  // window width
int win_h = 0;  // window height

void render() {
  
  win_w = window->mouse.x;
  win_h = window->mouse.y;
  
  double scale = fmin(win_w/(double)vp_w, win_h/(double)vp_h);
  
  int w = vp_w * scale;
  int h = vp_h * scale;
  
  // win
  S2D_DrawQuad(0, 0, 0.5, 0.5, 0.5, 1,
               win_w, 0, 0.5, 0.5, 0.5, 1,
               win_w, win_h, 0.5, 0.5, 0.5, 1,
               0, win_h, 0.5, 0.5, 0.5, 1);
  
  // vp
  S2D_DrawQuad(0, 0, 0.8, 0.8, 0.8, 1,
               w, 0, 0.8, 0.8, 0.8, 1,
               w, h, 0.8, 0.8, 0.8, 1,
               0, h, 0.8, 0.8, 0.8, 1);
}

int main() {
  
  window = S2D_CreateWindow(
    "Hello Viewport", 800, 600, NULL, render, 0
  );
  
  S2D_Show(window);
  return 0;
}
```

**Comment by blacktm** [comment]
New example with a centered viewport:

```c
#include <simple2d.h>
#include <math.h>

S2D_Window *window;
int vp_w = 50;
int vp_h = 50;
int win_w = 0;
int win_h = 0;

void render() {
  
  win_w = window->mouse.x;
  win_h = window->mouse.y;
  
  double scale = fmin(win_w/(double)vp_w, win_h/(double)vp_h);
  
  int w = vp_w * scale;
  int h = vp_h * scale;
  
  // win
  S2D_DrawQuad(0, 0, 0.5, 0.5, 0.5, 1,
               win_w, 0, 0.5, 0.5, 0.5, 1,
               win_w, win_h, 0.5, 0.5, 0.5, 1,
               0, win_h, 0.5, 0.5, 0.5, 1);
  
  // center the viewport
  int off_w = win_w/2 - w/2;
  int off_h = win_h/2 - h/2;
  
  S2D_DrawQuad(off_w, off_h, 0.8, 0.8, 0.8, 1,
               w + off_w, off_h, 0.8, 0.8, 0.8, 1,
               w + off_w, h + off_h, 0.8, 0.8, 0.8, 1,
               off_w, h + off_h, 0.8, 0.8, 0.8, 1);
}

int main() {
  
  window = S2D_CreateWindow(
    "Hello Triangle", 800, 600, NULL, render, S2D_RESIZABLE
  );
  
  S2D_Show(window);
  return 0;
}
```

**Comment by blacktm** [comment]
This has been implemented for GL2 and GL3 in 6430dab.

**Comment by blacktm** [comment]
Now that we have GLES support (6c98c0e3ab8c5b1b7649158ec073859abec9fc5a), this is working everywhere and fully implemented.
